### DISCUSSION TITLE
Is there a SwaggerUI alternative for gRPC?

### DISCUSSION THREAD

**Discussion by JohnLKramer** [grpc/grpc-java#11755] [General]
I knew that there are a lot of client tools that I could use but I'd like to just be able to add a dependency and some configurations and have a web page exposed for gRPC services. Does anything like that exist?

**Comment by conorheffron** [comment]
@JohnLKramer 

Googles **[gnostic](https://github.com/google/gnostic)** offers alternative with the **Open API app / plugin**:
- https://github.com/google/gnostic/tree/main/cmd/protoc-gen-openapi (not sure if still supported though)

**Note:** As an aside, _Open API_ can be used in conjunction with Swagger in `Java / Spring Boot` & `Python / Django` Applications with exposed web page.

### _Examples:_
#### 1. Java Based Service with Swagger
- https://www.ironoc.net/swagger-ui/index.html (OpenAPI / Swagger v3 with Spring Boot)
- https://github.com/conorheffron/ironoc/blob/main/src/main/java/net/ironoc/portfolio/controller/GitProjectsController.java (See `@Operation` & `@ApiResponses` annotations)
- https://github.com/conorheffron/ironoc/pull/69/files (setup config / dependencies)

#### 2. Python Based API with Swagger
- https://booking-sys-ebgefrdmh3afbhee.northeurope-01.azurewebsites.net/api/docs/ (Python Based Service with Django)
- https://github.com/conorheffron/booking-sys/blob/main/backend/hr/views.py (See `@extend_schema` decorators from Django Rest Framework (DRF) Spectacular)

#### 3. Try GraphQL 
 - Use Graph based API to document API on the fly with playground using query/mutation definitions in Schema
 - Click `Documentation Explorer`on the top left corner at [iRonoc.net GraphQL Playground](https://ironoc.net/graphiql?path=/graphql)
 - https://github.com/conorheffron/ironoc/blob/main/src/main/resources/graphql/ironocGraphSchema.graphql (GraphQL schema)

<img width="2505" height="519" alt="image" src="https://github.com/user-attachments/assets/0cd09586-7f27-4b26-ad01-2aff020b6373" />

#### 4. If app is AI driven, then might be worth exploring MCP server options for auto-generated doc
 - https://github.com/microsoft/mcp
